src/string.cr

```diff
@@ -4,6 +4,8 @@ require "c/string"
   require "crystal/iconv"
 {% end %}
 
+require "regex/match_data"
```

Comment:
IIRC there was a comment somewhere saying we shouldn't do this, I forgot where

Reply:
Yes, `string` is part of the core lib, `regex/match_data` is not. With that require, `regex/matchdata` would become part of core lib by extension which it shouldn't.